That required a good number of changes to satisfy the doctests and clippy. I suspect the bump in version is to blame.
I blindly listened to clippy, though the changes seemed to make sense. For the doc tests essentially every test got the same error message:
```rust
error[E0599]: `Peripherals` is not an iterator
  --> hal/src/tc/mod.rs:25:34
   |
11 | let pac = hal::pac::Peripherals::take().unwrap();
   |                                  ^^^^ `Peripherals` is not an iterator
   |
  ::: /mnt/c/Users/martin.mortsell/Documents/atsams70-rust/target/x86_64-unknown-linux-gnu/debug/build/atsamv71q21b-90d0014a7c81a20a/out/lib.rs:1:55241
   |
1  | ...All the peripherals."] # [allow (non_snake_case)] pub struct Peripherals { # [doc = "ACC"] pub ACC : ACC , # [doc = "AES"] pub AES : AES , # [doc = "AFEC0"] pub AF...
   |                                                      ---------------------- doesn't satisfy `Peripherals: Iterator`
   |
   = note: the following trait bounds were not satisfied:
           `Peripherals: Iterator`
           which is required by `&mut Peripherals: Iterator`

error: aborting due to previous error
```
I don't understand what the issue is, but solved it by changing all occurrences of `Peripherals::take()` to `Peripherals::steal()`. This change is not visible in the docs as those lines are hidden anyway.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,7 +19,7 @@
 - Examples now build and link again
 - [#62] Remove ambiguous reexports from `src/serial/mod.rs`.
 - TWIHS: Fix issue with clock frequency calculation.
-- Fix CI
+- Fix CI: source of PACs matrix + clippy satisfaction
 
 ## [v0.4.2] 2022-11-06
 

